internal/telemetry/probe.go

```diff
@@ -51,9 +58,11 @@ func (sp *statefulProbe) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 					logger.WithError(err).Warningf("%s health check failed. The server will terminate if this continues to happen.", HealthCheckEndpoint)
 				}
 				http.Error(w, err.Error(), http.StatusServiceUnavailable)
+				IncrementCounter(req.Context(), mReadinessProbes, tag.Insert(successKey, "false"))
 				return
 			}
 		}
+		IncrementCounter(req.Context(), mReadinessProbes, tag.Insert(successKey, "true"))
 		old := atomic.SwapInt32(sp.healthState, healthStateHealthy)
```

Comment:
It looks like ServeHTTP will respond "ok" even if there is no probe handler defined for a server. It is the expected behavior?

Reply:
If the readiness probe list is empty the behavior is to return ok. That's intentional, otherwise Kubernetes will kill the pod after a few failed calls.